src/serializer/ResidualHeapSerializer.js

```diff
@@ -551,8 +553,15 @@ export class ResidualHeapSerializer {
   }
 
   _serializeValueIntrinsic(val: Value): BabelNodeExpression {
-    invariant(val.intrinsicName);
-    return this.preludeGenerator.convertStringToMember(val.intrinsicName);
+    let intrinsicName = val.intrinsicName;
+    invariant(intrinsicName);
+    let intrinsicId = t.identifier(this.valueNameGenerator.generate(intrinsicName));
+    let declar = t.variableDeclaration("var", [
+      t.variableDeclarator(intrinsicId, this.preludeGenerator.convertStringToMember(intrinsicName)),
+    ]);
+    if (val.generator) this.emitter.emit(declar);
```

Comment:
If the generator present, the code in ResidualHeapVisitor will put the value into the scope of that generator, and then the emitter will in turn put the declaration into the code-block associated with that generator. I can add an invariant to that end here, so that you see the tie-back to the generator.